src/blockchainTx/web3/ContractsService.ts

```diff
@@ -0,0 +1,215 @@
+            data: hashed,
+            uri: '',
+        };
+        const attestation_2 = {
+            claimType: claimType_2,
+            scheme: 1,
+            issuer:  config.ownerAddress,
+            signature: signed2.signature,
+            data: hashed2,
+            uri: '',
+        };
+        this.ecosystemContract.methods.createNewUser(
+            newUser.userAddress, newUser.recoveryAddress,
+            [ attestation_1.claimType, attestation_2.claimType ],
+            [ attestation_1.issuer, attestation_2.issuer ],
+            attestation_1.signature + attestation_2.signature.slice(2),
+            attestation_1.data + attestation_2.data.slice(2),
+            [32, 32]).send({from: config.ownerAddress,
+            gas: 3000000,
+            gasPrice: '1'})
+            .then(function(receipt){
+                console.log(receipt);
+            });
+    }
+
+    async createNewCentralizedUser(newUser: CentralizedBlockchainUserDto) {
+        const data_text_1 = 'Verified Social';
+        const data_text_2 = 'Verified Kyc';
+        const dataHash_1 = Web3.utils.asciiToHex(data_text_1);
+        const dataHash_2 = Web3.utils.asciiToHex(data_text_2);
+        const predictAddress = await this.predictIdentityAddress(config.ecosystemAddress);
+        const claimType_1 = 1;
+        const hashed = this.web3.utils.soliditySha3(predictAddress, claimType_1, dataHash_1);
+        const prvSigner1 = this.web3.utils.randomHex(32);
+        const signed = await this.web3.eth.accounts.sign(hashed, prvSigner1);
+        const claimType_2 = 2;
+        const prvSigner2 = this.web3.utils.randomHex(32);
+        const hashed2 = this.web3.utils.sha3(predictAddress, claimType_2, dataHash_2);
+        const signed2 = await this.web3.eth.accounts.sign(hashed2, prvSigner2);
+
+        const attestation_1 = {
+            claimType: claimType_1,
+            scheme: 1,
+            issuer: config.ownerAddress,
+            signature: signed.signature,
+            data: hashed,
+            uri: '',
+        };
+        const attestation_2 = {
+            claimType: claimType_2,
+            scheme: 1,
+            issuer:  config.ownerAddress,
+            signature: signed2.signature,
+            data: hashed2,
+            uri: '',
+        };
+        this.ecosystemContract.methods.createNewCentralizedUser(
+            newUser.userAddress,
+            [ attestation_1.claimType, attestation_2.claimType ],
+            [ attestation_1.issuer, attestation_2.issuer ],
+            attestation_1.signature + attestation_2.signature.slice(2),
+            attestation_1.data + attestation_2.data.slice(2),
+            [32, 32]).send({from: config.ownerAddress,
+            gas: 3000000,
+            gasPrice: '1'})
+            .then(function(receipt){
+                console.log(receipt);
+            });
+    }
+
+    async predictIdentityAddress(wallet) {
+        const nonce = await new Promise(resolve => {
```

Comment:
Is it really required to wrap `getTransactionCount` with Promise?

Reply:
Think that getTransactionCount needs to query for the transaction block # of something on the network and for that reason its async. Is there another way that this should be programmed? This code was copied from an outside source and the team specified this is the best way to do it, its basically a trick where we predict what the next smart contract address will be based on the past transaction count of the wallet.